### PR TITLE
chore: Move compare mode sublayers to redux

### PR DESCRIPTION
**Pull request by MaximKYZ**
@belom88 Please review

### CODE REVIEW

File: src/redux/slices/flattened-sublayers-slice.tsx [hunk 1]
```diff
@@ -32,54 +42,118 @@ const flattenedSublayersSlice = createSlice({
...
     },
   },
+
   extraReducers: (builder) => {
     builder
-      .addCase(
-        getFlattenedSublayers.pending,
-        (state: flattenedSublayersState) => {
-          state.layerCounter += 1;
+      .addCase(getFlattenedSublayers.pending, (state, action) => {
```
Comment (belom88) [inline src/redux/slices/flattened-sublayers-slice.tsx:92]: could you add types? meta.arg.side is not clear

File: src/redux/slices/flattened-sublayers-slice.tsx [hunk 2]
```diff
@@ -32,54 +42,118 @@ const flattenedSublayersSlice = createSlice({
...
+        if (action.meta.arg.side === ComparisonSideMode.left) {
+          state.left.layerCounter += 1;
+        } else if (action.meta.arg.side === ComparisonSideMode.right) {
+          state.right.layerCounter += 1;
+        } else {
+          state.single.layerCounter += 1;
         }
```
Comment (belom88) [inline src/redux/slices/flattened-sublayers-slice.tsx:99]:
```suggestion
        const side = action.meta.arg.side || 'single';
        state[side].layerCounter++;
```

File: src/redux/slices/flattened-sublayers-slice.tsx [hunk 3]
```diff
@@ -32,54 +42,118 @@ const flattenedSublayersSlice = createSlice({
...
+            action.payload.side === ComparisonSideMode.right &&
+            action.payload.sublayers.layerCounter === state.right.layerCounter
+          ) {
+            return { ...state, right: action.payload.sublayers };
+          } else if (
+            !action.payload.side &&
+            action.payload.sublayers.layerCounter === state.single.layerCounter
+          ) {
+            return { ...state, single: action.payload.sublayers };
           }
```
Comment (belom88) [inline src/redux/slices/flattened-sublayers-slice.tsx:125]:
```suggestion
          const side = action.payload.side || 'single';
          if (action.payload.sublayers.layerCounter === state[side].layerCounter) {
            return { ...state, [side]: action.payload.sublayers };
          }
```

File: src/redux/slices/flattened-sublayers-slice.tsx [hunk 4]
```diff
@@ -32,54 +42,118 @@ const flattenedSublayersSlice = createSlice({
...
+          if (action.payload.index in state.right.layers) {
+            destLayer = state.right.layers[action.payload.index];
+          }
+          break;
+        default:
+          destLayer = state.single.layers[action.payload.index];
+      }
+      if (destLayer) {
+        destLayer.visibility = action.payload.visibility;
       }
```
Comment (belom88) [inline src/redux/slices/flattened-sublayers-slice.tsx:86]:
```suggestion
      const side = action.payload.side || 'single';
      const layers = state[side].layers;
      if (action.payload.index in layers) {
        layers.visibility = action.payload.visibility;
      }
```

File: src/redux/slices/flattened-sublayers-slice.tsx [hunk 5]
```diff
@@ -32,54 +42,118 @@ const flattenedSublayersSlice = createSlice({
...
+      switch (action.payload.side) {
+        case ComparisonSideMode.left:
+          state.left.layers = action.payload.layers;
+          break;
+        case ComparisonSideMode.right:
+          state.right.layers = action.payload.layers;
+          break;
+        default:
+          state.single.layers = action.payload.layers;
+      }
```
Comment (belom88) [inline src/redux/slices/flattened-sublayers-slice.tsx:59]:
```suggestion
      state[action.payload.side || 'single'].layers = action.payload.layers;
```

File: src/redux/slices/flattened-sublayers-slice.tsx [hunk 6]
```diff
@@ -4,26 +4,36 @@ import {
...
 } from "../../types";
 import { I3SBuildingSceneLayerLoader } from "@loaders.gl/i3s";
 import { buildSublayersTree } from "../../utils/sublayers";
 import { load } from "@loaders.gl/core";
 import { RootState } from "../store";
 
-// Define a type for the slice state
-interface flattenedSublayersState {
+// Define a type for the slice states
+type sublayersState = {
```
Comment (belom88) [inline src/redux/slices/flattened-sublayers-slice.tsx:15]: Types must start from upper case
```suggestion
type SublayersState = {
```

**Review by belom88 — CHANGES_REQUESTED** [review]
We also have to add tests for slices in a separate PR

File: src/redux/slices/flattened-sublayers-slice.tsx (hunk 1 above)
Comment (MaximKYZ) [reply src/redux/slices/flattened-sublayers-slice.tsx:92]: As per redux's TypeScript usage guide it's already typed - https://redux.js.org/usage/usage-with-typescript#typing-extrareducers

File: src/redux/slices/flattened-sublayers-slice.tsx (hunk 2 above)
Comment (MaximKYZ) [reply src/redux/slices/flattened-sublayers-slice.tsx:99]: Done

File: src/redux/slices/flattened-sublayers-slice.tsx (hunk 3 above)
Comment (MaximKYZ) [reply src/redux/slices/flattened-sublayers-slice.tsx:125]: Done

File: src/redux/slices/flattened-sublayers-slice.tsx (hunk 4 above)
Comment (MaximKYZ) [reply src/redux/slices/flattened-sublayers-slice.tsx:86]: Done

File: src/redux/slices/flattened-sublayers-slice.tsx (hunk 5 above)
Comment (MaximKYZ) [reply src/redux/slices/flattened-sublayers-slice.tsx:59]: Done

File: src/redux/slices/flattened-sublayers-slice.tsx (hunk 6 above)
Comment (MaximKYZ) [reply src/redux/slices/flattened-sublayers-slice.tsx]: Done

File: src/redux/slices/flattened-sublayers-slice.tsx [hunk 7]
```diff
@@ -32,54 +42,82 @@ const flattenedSublayersSlice = createSlice({
...
+    switch (side) {
+      case ComparisonSideMode.left:
+        currentLayer = state.flattenedSublayers.left.layerCounter;
+        break;
+      case ComparisonSideMode.right:
+        currentLayer = state.flattenedSublayers.right.layerCounter;
+        break;
+      default:
+        currentLayer = state.flattenedSublayers.single.layerCounter;
+    }
```
Comment (belom88) [inline src/redux/slices/flattened-sublayers-slice.tsx:120]:
```suggestion
    const currentLayer = state.flattenedSublayers[side || 'single'].layerCounter;
```

**Review by belom88 — APPROVED** [review]

File: src/redux/slices/flattened-sublayers-slice.tsx (hunk 7 above)
Comment (MaximKYZ) [reply src/redux/slices/flattened-sublayers-slice.tsx:120]: Done

**Pull request merged**